dotCMS/src/main/java/com/dotcms/services/VanityUrlServices.java

```diff
@@ -162,14 +163,25 @@ public void onCommitListener(CommitListenerEvent commitListenerEvent) {
         try {
             if (contentlet.isVanityUrl()) {
 
-                //When the contentlet finished to index we need to invalidate it on cache
-                boolean indexed = APILocator.getContentletAPI()
-                        .isInodeIndexed(contentlet.getInode(),
-                                contentlet.isLive(), contentlet.isWorking());
-                if (indexed) {
-                    //Invalidate this VanityURL
-                    VanityUrlServices.getInstance().invalidateVanityUrl(contentlet);
-                }
+                LocalTransaction.wrap(() -> {
+
+                    //When the contentlet finished to index we need to invalidate it on cache
+                    boolean indexed = false;
+                    try {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)